### DISCUSSION TITLE
Storj

### DISCUSSION THREAD

**Discussion by zimbres** [backube/volsync#144] [Q&A] [answered]
Is it possible to use VolSync with Storj?

**Comment by JohnStrunk** [comment]
Maybe...

Rclone appears to support Tardigrade (https://rclone.org/tardigrade/), so it may be possible.
If you can create an `rclone.conf` that accesses your storage and can use `rclone sync`, there's a good chance you could get it working w/ VolSync.

If you want to give it a try:
* Take a look at the volsync rclone docs: https://volsync.readthedocs.io/en/stable/usage/rclone/index.html
* You're rclone.conf that is configured for Tardigrade will be what you use as the rclone Secret
* You will need to set `rcloneConfigSection` and `rcloneDestPath` appropriately in the ReplicationSource/ReplicationDestination
* The actual rclone commands that the VolSync mover executes are here: https://github.com/backube/volsync/blob/2a78c57773a4439a49d213e181fe2db171e20fc9/mover-rclone/active.sh#L23-L33


I'd love to hear if you get it working!

**Comment by RobotSail** [comment]
@zimbres Were you ever able to get this working? I'd also be curious to know

**Comment by RobotSail** [comment, marked as answer]
@zimbres Since Storj has S3-compatibility, you could use Storj with VolSync would be through [rclone](https://rclone.org/storj/). 


You could also set up an S3 bucket with a service such as [Filebase](https://filebase.com/) and select Storj as your storage backend. This should work with any data mover that can be setup with an S3-compatible API, including rclone. You would also be able to use additional backends including Sia, IPFS, and Skynet, which are not directly supported through Rclone (yet).

**Reply by zimbres** [reply]
Hi, I wasn't aware, but Storj supports S3 Api.

When generating an Access Token, in the last step there is an option to generate S3 credentials.

**Reply by RobotSail** [reply]
I see that now. It also looks like rclone has [direct support](https://rclone.org/storj/) for Storj.
